src/components/chart-preview/ChartPreview.tsx

```diff
@@ -71,6 +85,59 @@ export const ActualChart = ({
     return fullFigureNode;
   };
 
+  const onLegendClick = (e: any) => {
+    // check if the passed 'e.curveNumber' is in filteredColumnIndexes
+    // if it is, that means its filter is on and should be turned off
```

Comment:
I'm finding it tricky to keep track of whether 'filtered' means present or not present. These 2 lines of comment I think accentuate the point. Is it worth considering renaming it to 'hidden' or 'visible' so its clearer what the consequence of the filter action is? so `filteredColumnIndexes` might become `visibleColumnIndexes` or something? 

If not, I think at least in these comments it would be helpful to be clearer that `if it is, that means its filter is on and the column should not be visible` or whatever the appropriate consequence is